spec/src/main/asciidoc/metrics.asciidoc

```diff
@@ -0,0 +1,178 @@
+//
+// Copyright (c) 2018 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Andrew Rouse
+
+== Integration with Microprofile Metrics
+
+When Microprofile Fault Tolerance and Microprofile Metrics are used together, metrics are automatically added for each of
+the methods annotated with a `@Retry`, `@Timeout`, `@CircuitBreaker` or `@Bulkhead` annotation.
+
+=== Names
+
+The automatically added metrics follow a consistent pattern which includes the fully qualified name of the annotated method.
+In the tables below, the placeholder `<name>` should be replaced by the fully qualified method name with all `.` characters replaced with `_`.
+
+=== Metrics added for `@Retry`
```

Comment:
I think that also a metric for average number of retries per call would be useful to monitor stability of the call.

Reply:
With name `ft_<name>_retry_retries_average`

Reply:
Or the number of retries per call could be a histogram, with min, max and mean number of retries per call.

Reply:
You should be able to calculate the average retries per call from the other metrics since you can compute the total number of calls and we also expose the total number of retries.

Prometheus makes it fairly easy to do this sort of calculation and graph the results over time.